version: "3"
services:
  postgres:
    image: postgres:alpine
    restart: always
    networks:
      - back
    expose:
      - "5432"
    environment:
      POSTGRES_USER: pleroma
      POSTGRES_PASSWORD: pleroma
      POSTGRES_DB: pleroma
    volumes:
      - $DOCKER_DATADIR/db:/var/lib/postgresql/data
      - ./pleroma/initdb.sql:/docker-entrypoint-initdb.d/pleroma.sql

  pleroma:
    build: ./mypleroma
    image: pleroma
    restart: always
    command: mix phx.server
    networks:
      - back
      - front
    depends_on:
      - postgres
    expose:
      - "4000"
    volumes:
      - ./config:/pleroma/config
      - ./priv:/pleroma/priv
      - ./custom.d:/custom.d:ro
      - ./pleroma/config.exs:/home/pleroma/pleroma/config/prod.secret.exs:ro
      - ./pleroma/config.exs:/home/pleroma/pleroma/config/dev.secret.exs:ro
      - $DOCKER_DATADIR/uploads:/uploads

  nginx:
    build: ./nginx
    restart: always
    networks:
      - front
    depends_on:
      - pleroma
    ports:
      - "443:443"
      - "80:80"
    volumes:
      - "./certbot/conf:/etc/letsencrypt"
      - "./certbot/www:/var/www/certbot"
      - "./nginx/conf:/etc/nginx/conf.d/"

  certbot:
    image: certbot/certbot
    restart: always
    volumes:
      - "./certbot/conf:/etc/letsencrypt"
      - "./certbot/www:/var/www/certbot"

networks:
    front:
        external: true
    back:
        external: true
